modules/sdk-coin-atom/test/unit/transactionBuilder/StakingActivateBuilder.ts

```diff
@@ -0,0 +1,98 @@
+    should.equal(tx.type, TransactionType.StakingActivate);
+    (tx as AtomTransaction).atomTransaction.gasBudget.should.deepEqual(testTx.gasBudget);
+    const rawTx = tx.toBroadcastFormat();
+    should.equal(rawTx, testTx.signedTxBase64);
+  });
+
+  it('should build a WithdrawRewards tx without signature', async function () {
+    const txBuilder = factory.getStakingActivateBuilder();
+    txBuilder.sequence(testTx.sequence);
+    txBuilder.gasBudget(testTx.gasBudget);
+    txBuilder.messages([testTx.sendMessage.value]);
+    txBuilder.publicKey(toHex(fromBase64(testTx.pubKey)));
+    const tx = await txBuilder.build();
+    const json = await (await txBuilder.build()).toJson();
+    should.equal(tx.type, TransactionType.StakingActivate);
+    should.deepEqual(json.gasBudget, testTx.gasBudget);
+    should.deepEqual(json.sendMessages, [testTx.sendMessage]);
+    should.deepEqual(json.publicKey, toHex(fromBase64(testTx.pubKey)));
+    should.deepEqual(json.sequence, testTx.sequence);
+    tx.toBroadcastFormat();
+  });
+
+  it('should sign a WithdrawRewards tx', async function () {
+    const txBuilder = factory.getStakingActivateBuilder();
+    txBuilder.sequence(testTx.sequence);
+    txBuilder.gasBudget(testTx.gasBudget);
+    txBuilder.messages([testTx.sendMessage.value]);
+    txBuilder.accountNumber(testTx.accountNumber);
+    txBuilder.chainId(testTx.chainId);
+    txBuilder.sign({ key: toHex(fromBase64(testTx.privateKey)) });
+    const tx = await txBuilder.build();
+    const json = await (await txBuilder.build()).toJson();
+    should.equal(tx.type, TransactionType.StakingActivate);
+    should.deepEqual(json.gasBudget, testTx.gasBudget);
+    should.deepEqual(json.sendMessages, [testTx.sendMessage]);
+    should.deepEqual(json.publicKey, toHex(fromBase64(testTx.pubKey)));
+    should.deepEqual(json.sequence, testTx.sequence);
+    const rawTx = tx.toBroadcastFormat();
+    should.equal(tx.signature[0], toHex(fromBase64(testTx.signature)));
+    should.equal(rawTx, testTx.signedTxBase64);
+  });
+
+  xit('should submit a send transaction', async () => {
+    const keyPair = new KeyPair({ prv: toHex(fromBase64(testTx.privateKey)) });
+    const axiosConfig = {
+      headers: {
+        'Content-Type': 'application/cbor',
+      },
+      timeout: 10000,
+    };
+
+    const txBuilder = factory.getStakingActivateBuilder();
+    txBuilder.sequence(10);
+    txBuilder.gasBudget(testTx.gasBudget);
+    txBuilder.messages([testTx.sendMessage.value]);
+    txBuilder.accountNumber(testTx.accountNumber);
+    txBuilder.chainId(testTx.chainId);
+    txBuilder.sign({ key: keyPair.getKeys().prv });
+    const tx = await txBuilder.build();
+    const serializedTx = tx.toBroadcastFormat();
+
+    try {
+      const res = await axios.post(
+        'https://rest.sentry-01.theta-testnet.polypore.xyz/cosmos/tx/v1beta1/txs',
+        {
+          mode: 'BROADCAST_MODE_BLOCK',
+          tx_bytes: serializedTx,
+        },
+        axiosConfig
```

Comment:
and to confirm, when you submitted this, it was successful on chain?

Reply:
yes it was. https://explorer.theta-testnet.polypore.xyz/transactions/3D9583366D1D1AA82229800DE323F6B26E0170AD4CE610D19F08CD45D36505E2 this is the one I guess. Anyone can run it, only the sequence number needs to be changed